hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineUtils.java

```diff
@@ -308,6 +309,38 @@ public static HoodieTimeline getCommitsTimelineAfter(
     return timelineSinceLastSync;
   }
 
+  public static HoodieTimeline getCommitsTimeLineAfterByCompletionTimeRange(
+      HoodieTableMetaClient metaClient,
+      String exclusiveStartCompletionTime,
+      String inclusiveEndCompletionTime) {
+    HoodieDefaultTimeline writeTimeline = metaClient.getActiveTimeline().getWriteTimeline();
+
+    HoodieDefaultTimeline timeline;
+    if (writeTimeline.isBeforeTimelineStartsByCompletionTime(exclusiveStartCompletionTime)) {
+      // need to load archived timeline as well
+      try (CompletionTimeQueryView view = new CompletionTimeQueryView(metaClient)) {
+        List<String> instants = view.getStartTimes(
+            exclusiveStartCompletionTime,
+            inclusiveEndCompletionTime,
+            RangeType.OPEN_CLOSED);
+        if (instants.isEmpty()) {
+          LOG.warn("Did not find any instant when getting timeline, using write timeline for incremental read");
+          timeline = writeTimeline;
+        } else {
+          timeline = metaClient.getArchivedTimeline(instants.get(0))
+              .mergeTimeline(writeTimeline)
+              .mergeTimeline(writeTimeline);
```

Comment:
If this is still needed, only need to do one merge here.

Reply:
Yes, we only need to merge once here. Will fix this typo